### PR TITLE
feat: add require-jsdoc rule

### PR DESCRIPTION
**Pull request by ah-yu**
ref #970

### CODE REVIEW

File: src/rules/require_jsdoc.rs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  ) {
+    if let DefaultDecl::Fn(_) = _n.decl {
+      if !check_jsdoc_exsit(_ctx.leading_comments_at(_n.span().lo)) {
+        _ctx.add_diagnostic(_n.span(), CODE, MESSAGE);
+      }
+    }
+  }
+}
+
+fn check_jsdoc_exsit<'c>(
```
Comment (dsherret) [inline src/rules/require_jsdoc.rs:67]: typo: `check_jsdoc_exists`

**Review by dsherret — COMMENTED** [review]
@ah-yu it looks good so far, but I believe we need to do it for more than just functions? Also, I think perhaps this should also look at public members of classes/interfaces?

**Comment by ah-yu** [comment]
added interfaces/classes jsdoc comment checks.
`eslint-plugin-jsdoc/require-jsdoc` only checks `FunctionDeclaration` by default, should we check classes/interfaces by default? 
ref: https://github.com/gajus/eslint-plugin-jsdoc#require

**Comment by bartlomieju** [comment]
@ah-yu the PR looked okay, what's the reason to close it?

**Comment by bartlomieju** [comment]
@dsherret please review, should we add more feature before landing?

**Comment by ah-yu** [comment]
@bartlomieju Sorry for late response, I was kind of busy those days so I closed this PR. And I'd like to finish it.

**Comment by iuioiua** [comment]
Hi @ah-yu, is there any way we could progress this PR? This change could be valuable for modules that'd like to enforce documentation.

**Comment by sigmaSd** [comment]
It would be definitely great to have something like rust `#![warn(missing_docs)]` hopefully this gets picked up again, its really useful for library authors (and in my opinion it should be opt in like rust's)

**Comment by dsherret** [comment]
Thanks, but we ended up going with a much more powerful linting system as part of `deno doc`. For details, see this PR https://github.com/denoland/deno/pull/21032

**Pull request closed without merging**